song-server/src/main/java/bio/overture/song/server/service/ValidationService.java

```diff
@@ -98,59 +98,31 @@ public Optional<String> validate(@NonNull JsonNode payload) {
       if (analysisType.getOptions() != null && analysisType.getOptions().getFileTypes() != null) {
         fileTypes = analysisType.getOptions().getFileTypes();
       }
-
-      if(Objects.isNull(analysisType.getVersion())){
-        // newest version is fetched
-
+      // Only use the file type rules from the corresponding schema version
+      Integer schemaVersion = analysisType.getVersion();
+      if (Objects.isNull(schemaVersion)) {
```

Comment:
This section should not be necessary. The call to `analysisTypeService.getAnalysisType(analysisTypeId, false);` will get the latest version if none is provided, and if the schema can't be found it will throw an error. We should just be able to move onto the logic on line 105 where we validate fileTypes.